### PR TITLE
2303/edit subtasks/handle save subtasks

### PR DESCRIPTION
**Pull request by aleneagu96**

### CODE REVIEW

**Review by andregeorgi — APPROVED** [review]

File: src/components/Modal/Modal.jsx [hunk 1]
```diff
@@ -63,6 +63,12 @@ const Modal = (props) => {
     );
   };
 
+  const handleSaveClick = () => {
+    const updatedSubtasks = subtaskText.filter((subtask) => subtask !== "");
+    handleSave(todoText, priority, updatedSubtasks);
+
```
Comment (sorincretescu) [inline src/components/Modal/Modal.jsx:69]: Wouldn't this look better without the useless new line? ^^

File: src/components/Modal/Modal.jsx [hunk 2]
```diff
@@ -91,7 +97,7 @@ const Modal = (props) => {
         {subtasks?.map((subtask, index) => (
           <Input
             key={index}
-            value={subtask}
+            value={subtaskText[index]}
```
Comment (sorincretescu) [inline src/components/Modal/Modal.jsx:100]: Can you please explain this? Where is it coming from? I see no "subtaskText".

File: src/views/todo/Todo.jsx [hunk 3]
```diff
@@ -83,13 +84,20 @@ const Todo = () => {
     setOpenEditModal(false);
   };
 
-  const handleSave = (description, priority) => {
+  const handleSave = (description, priority, editedSubtasks) => {
     const newTodos = [...todos];
     newTodos[currentTodo].description = description;
     newTodos[currentTodo].priority = priority;
+    newTodos[currentTodo].subtasks = editedSubtasks;
```
Comment (sorincretescu) [inline src/views/todo/Todo.jsx:91]: Another way of writing this would be newTodos[currentTodo] = { priority, description, subtasks: editedSubtasks };

**Review by sorincretescu — CHANGES_REQUESTED** [review]
Please, when you solve comments, try to understand what and why is happening. Especially with suggestions.

File: src/components/Modal/Modal.jsx (hunk 1 above)
Comment (aleneagu96) [reply src/components/Modal/Modal.jsx]: done

File: src/views/todo/Todo.jsx (hunk 3 above)
Comment (aleneagu96) [reply src/views/todo/Todo.jsx]: done

**Review by andregeorgi — APPROVED** [review]

**Review by sorincretescu — APPROVED** [review]

**Pull request merged**